lib/cocoapods/installer/analyzer.rb

```diff
@@ -421,23 +426,21 @@ def generate_targets(resolver_specs_by_target, target_inspections)
       # @return [AggregateTarget]
       #
       def generate_target(target_definition, target_inspections, pod_targets, resolver_specs_by_target)
+        target_requires_64_bit = requires_64_bit_archs?(target_definition.platform)
         if installation_options.integrate_targets?
           target_inspection = target_inspections[target_definition]
           raise "missing inspection: #{target_definition.name}" unless target_inspection
           user_project = target_inspection.project
           client_root = user_project.path.dirname.realpath
           user_target_uuids = target_inspection.project_target_uuids
           user_build_configurations = target_inspection.build_configurations
-          archs = target_inspection.archs
+          archs = target_requires_64_bit ? ['$(ARCHS_STANDARD_64_BIT)'] : target_inspection.archs
         else
           user_project = nil
           client_root = config.installation_root.realpath
           user_target_uuids = []
           user_build_configurations = target_definition.build_configurations || Target::DEFAULT_BUILD_CONFIGURATIONS
-          archs = []
-          if target_definition.platform && target_definition.platform.name == :osx
-            archs = ['$(ARCHS_STANDARD_64_BIT)']
-          end
+          archs = target_requires_64_bit ? ['$(ARCHS_STANDARD_64_BIT)'] : []
```

Comment:
The previous implementation always included it for `:osx` platforms, but the integration spec PR has changes for macOS targets - any insights? It seems like the logic is correct here but the integration specs are making me doubt myself

Reply:
weird I have no idea but looking quickly as to why.

Reply:
@amorde will try to check it out and diagnose it myself.

Reply:
ok, I haven't have much time to take another look at this unfortunately